src/routes_admin.py

```diff
@@ -0,0 +1,243 @@
+            new_student_id = request.form.get('netid')
+            if new_student_id is None:
+                return util.error("Cannot find netid field")
+            result = db.add_student_to_course(cid, str(new_student_id))
+            if none_modified(result):
+                return util.error(f"'{new_student_id}' is already a student")
+            return util.success(f"Successfully added {new_student_id}")
+
+        @blueprint.route("/staff/course/<cid>/remove_student", methods=["POST"])
+        @auth.require_auth
+        @auth.require_admin_status
+        def remove_course_student(netid, cid):
+            netid = request.form.get('netid')
+            result = db.remove_student_from_course(cid, netid)
+            if none_modified(result):
+                return util.error(f"'{netid}' is not a student")
+            return util.success(f"Successfully removed '{netid}'")
+
+        @blueprint.route("/staff/course/<cid>/<aid>/", methods=["POST"])
+        @auth.require_auth
+        @auth.require_admin_status
+        def staff_add_assignment(netid, cid, aid):
+            missing = util.check_missing_fields(request.form,
+                                                *["max_runs", "quota", "start", "end", "config", "visibility"])
+            if missing:
+                return util.error(f"Missing fields ({', '.join(missing)}).")
+
+            if not util.valid_id(aid):
+                return util.error("Invalid Assignment ID. Allowed characters: a-z A-Z _ - .")
+
+            new_assignment = db.get_assignment(cid, aid)
+            if new_assignment:
+                return util.error("Assignment ID already exists.")
+
+            try:
+                max_runs = int(request.form["max_runs"])
+                if max_runs < MIN_PREDEADLINE_RUNS:
+                    return util.error(f"Max Runs must be at least {MIN_PREDEADLINE_RUNS}.")
+            except ValueError:
+                return util.error("Max Runs must be a positive integer.")
+
+            quota = request.form["quota"]
+            if not db.Quota.is_valid(quota):
+                return util.error("Quota Type is invalid.")
+
+            start = util.parse_form_datetime(request.form["start"]).timestamp()
+            end = util.parse_form_datetime(request.form["end"]).timestamp()
+            if start is None or end is None:
+                return util.error("Missing or invalid Start or End.")
+            if start >= end:
+                return util.error("Start must be before End.")
+
+            try:
+                config = json.loads(request.form["config"])
+                msg = bw_api.set_assignment_config(cid, aid, config)
+
+                if msg:
+                    return util.error(f"Failed to add assignment to Broadway: {msg}")
+            except json.decoder.JSONDecodeError:
+                return util.error("Failed to decode config JSON")
+
+            visibility = request.form["visibility"] == "visible"
```

Comment:
This is probably an artifact of css visibility settings having the following options:
```
visibility: visible|hidden|collapse|initial|inherit;
```

It might be better to leave as is.

Reply:
It seems the "visible" and "hidden" are two string values hard coded into the input radio. There could be better ways of doing this, such as instead of two radios, we can have one "on/off" switch. But it's not as trivial to change this field from a string to a boolean.

Reply:
Gotcha, we can leave it as it is for now then.